client/src/main/proto/app/cash/backfila/client_service.proto

```diff
@@ -98,14 +98,24 @@ message RunBatchRequest {
 
   // Provided data from the source backfill if this is the target backfill in a pipelined backfill.
   optional PipelinedData pipelined_data = 7;
+
+  // Provided data from the backfila client allowing this request to continue this batch.
+  optional bytes partial_batch_bookmark = 8;
 }
 message RunBatchResponse {
   // How much to sleep before starting new batches. For example can be set to replication lag.
   optional uint64 backoff_ms = 1;
 
   optional PipelinedData pipelined_data = 2;
 
+  // Stacktrace that will be surfaced to the user.
+  // If this is provided this batch request is assumed to have failed.
   optional string exception_stack_trace = 3;
+
+  // Provided data from the backfila client telling the next invocation how to continue.
+  // Allows the client to return even if a batch is not done.
+  // If this is provided this batch request is considered incomplete.
+  optional bytes partial_batch_bookmark = 8;
```

Comment:
We had hack week last week which is why it is taking a while to get back to this.  Correct me if I'm wrong but I think you are suggesting:

```
  optional KeyRange remaining_range = 8;
}
```

It would save us from having this in the Request. I think I really like it?

So in the Dynamo case there will be extra data in this case and I wonder if it makes sense to store this data with the KeyRange data.

It might just make sense since this extra data is giving us information about what is left/done.

Reply:
yea that's what I'm thinking

Reply:
Finally got this done. That test took a while. Turns out I needed to delay a few times to make sure all the other coroutines were done...